includes/Checker/CLI_Runner.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\CLI_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * CLI Runner class.
+ *
+ * @since n.e.x.t
+ */
+class CLI_Runner extends Abstract_Check_Runner {
+
+	/**
+	 * Checks if the current request is a CLI request for the Plugin Checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	public function is_plugin_check() {
+		if ( empty( $_SERVER['argv'] ) ) {
+			return false;
+		}
+
+		if (
+			'wp' === $_SERVER['argv'][0] &&
+			'plugin' === $_SERVER['argv'][1] &&
+			'check' === $_SERVER['argv'][2]
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the plugin main file based on the request.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The absolute path to the plugin main file.
+	 */
+	private function get_plugin_main_file() {
+		// Get the plugin name from the command line arguments.
+		$plugin_name = $_SERVER['argv'][3];
+
+		return WP_PLUGIN_DIR . '/' . $plugin_name . '/' . $plugin_name . '.php';
```

Comment:
This is unreliable, there is no guarantee that the plugin's main file has the same name as the plugin directory.

For reference, see https://github.com/felixarntz/plugin-check/blob/main/includes/CLI/Plugin_Check_Command.php#L189: This method supports both a plugin slug or a full plugin basename being passed, and it only works if that plugin is actually installed.

Maybe introduce a class like `Plugin_Request_Utility` with a static method `get_plugin_basename_from_input( $plugin_slug_or_basename )`? That way we could use that logic here, but also later in the actual CLI command, the `AJAX_Runner`, etc.

All of that also means that this _may_ fail, if some invalid slug or basename is passed, in which case we should probably throw an exception. So we should probably clarify in this method, and also more importantly in the `get_checks()` method where this private method is called.

Reply:
Thanks @felixarntz I've added the `Plugin_Request_Utility` class and tests